### ISSUE TITLE
Ngrx practice on Angular Project

### ISSUE THREAD

**Issue by reboottime** [reboottime/Crash-Angular-For-Interview#11]

**Comment by reboottime** [comment]
## Rationale 

![image](https://github.com/reboottime/Crash-Angular-For-Interview/assets/84222273/976429fd-0ab4-4dc0-a781-466ecdc2bcaf)

Vs

Redux diagram

<img width="854" alt="image" src="https://github.com/reboottime/Crash-Angular-For-Interview/assets/84222273/79a17703-0b8f-4e4e-b215-54faad3c24b9">


The very basic [sample code pattern](https://ngrx.io/generated/live-examples/store/stackblitz.html)

**Comment by reboottime** [comment]
## Example variations

- For no external interaction  example, (.ie, http request)  [sample code pattern](https://ngrx.io/generated/live-examples/store/stackblitz.html)
- For example with external interaction

**Comment by reboottime** [comment]
## State level

In NgRx, `StoreModule.forRoot()` and `StoreModule.forFeature()` are methods used to configure the state management for an application. They serve different purposes:

1. **`StoreModule.forRoot()`**:

   - This method is used to set up the root application state. It should be called only once in your application, typically in the `AppModule`.
   - It configures the global state for your application and initializes the top-level state slices.
   - The first argument to `forRoot()` is the initial state of the application, and the second argument is a configuration object (optional).
   - Example usage:

     ```typescript
     import { StoreModule } from '@ngrx/store';

     @NgModule({
       imports: [
         // ...
         StoreModule.forRoot({ /* initial state */ }),
         // ...
       ],
       // ...
     })
     export class AppModule { }
     ```

2. **`StoreModule.forFeature()`**:

   - This method is used to set up feature modules. Feature modules are used to organize the state of specific features or parts of your application.
   - It is called once for each feature module and allows you to add additional state slices to your application.
   - Example usage:

     ```typescript
     import { StoreModule } from '@ngrx/store';

     @NgModule({
       imports: [
         // ...
         StoreModule.forFeature('featureName', /* feature reducer */),
         // ...
       ],
       // ...
     })
     export class FeatureModule { }
     ```

   - In the above example, `'featureName'` is a unique string identifier for this feature module, and `/* feature reducer */` is the reducer function for this feature's state slice.

   - You can use `createFeatureSelector` and `createSelector` to access the state of the feature module.

3. **When to Use Each**:

   - `forRoot`: Use this in the root module of your application (typically `AppModule`) to set up the global state for your application. It initializes the top-level state slices and should only be called once.

   - `forFeature`: Use this in feature modules to add additional state slices. Feature modules are typically used to manage the state for specific parts of your application. You can call `forFeature` multiple times to set up state for different features.

In summary, `forRoot` is used in the root module to configure the global application state, while `forFeature` is used in feature modules to manage specific parts of the application's state. This helps in organizing and modularizing the state management in NgRx applications.
